arm9/src/unk_02031734.c

```diff
@@ -0,0 +1,2371 @@
+{
+    if (UNK_021C5A00 != NULL)
+    {
+        if (UNK_021C5A00->unk24 == FUN_02032E8C)
+        {
+            return 1;
+        }
+
+        if (UNK_021C5A00->unk24 == FUN_02032E90 && UNK_021C5A00->unk44 != 0)
+        {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+THUMB_FUNC void FUN_02033288(struct SaveBlock2 *sav2)
+{
+    if (UNK_021C5A00 == NULL)
+    {
+        FUN_02016828(3, 0xf, 0x100);
+        UNK_021C5A00 =
+            (struct UnkStruct_02031734 *)AllocFromHeap(0xf, sizeof(struct UnkStruct_02031734));
+        MI_CpuFill8(UNK_021C5A00, 0, sizeof(struct UnkStruct_02031734));
+
+        UNK_021C5A00->unk3F = 0x18;
+        UNK_021C5A00->unk46 = 1;
+        UNK_021C5A00->unk28 = sav2;
+        FUN_020334E8(0, 1);
+        FUN_020166A8(1);
+    }
+}
+
+THUMB_FUNC void FUN_020332DC()
+{
+    if (UNK_021C5A00 != 0)
+    {
+        FUN_020166B8(1);
+        FUN_020334E8(0, 0);
+        FreeToHeap(UNK_021C5A00);
+        UNK_021C5A00 = NULL;
+        FUN_020168D0(0xf);
+    }
+}
+
+THUMB_FUNC void FUN_02033310(struct SaveBlock2 *sav2)
+{
+    if (UNK_021C5A00 == NULL)
+    {
+        FUN_02016828(3, 0xf, 0x100);
+        UNK_021C5A00 =
+            (struct UnkStruct_02031734 *)AllocFromHeap(0xf, sizeof(struct UnkStruct_02031734));
+        MI_CpuFill8(UNK_021C5A00, 0, sizeof(struct UnkStruct_02031734));
+        UNK_021C5A00->unk3F = 0x19;
+        UNK_021C5A00->unk46 = 1;
+        UNK_021C5A00->unk28 = sav2;
+        FUN_020334E8(0, 1);
+        FUN_020166A8(1);
+    }
+}
+
+THUMB_FUNC void FUN_02033364()
+{
+    if (UNK_021C5A00 != 0)
+    {
+        FUN_020166B8(1);
+        FUN_020334E8(0, 0);
+        FreeToHeap(UNK_021C5A00);
+        UNK_021C5A00 = NULL;
+        FUN_020168D0(0xf);
+    }
+}
+
+THUMB_FUNC u32 FUN_02033398()
+{
+    u32 st4;
+    u32 st0;
+    if (UNK_021C5A00 == NULL)
+    {
+        return 0;
+    }
+
+    if (UNK_021C5A00->unk3F == 0x18)
+    {
+        if (MOD04_021DD718(&st4, &st0) != 0)
+        {
+            MOD04_021DD6F0();
+            return 1;
+        }
+    }
+    else if (UNK_021C5A00->unk3F == 0x19)
+    {
+        if (MOD04_021DD718(&st4, &st0) != 0 && st0 == 7)
+        {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+THUMB_FUNC void FUN_020333F0()
+{
+    s32 res = MOD04_021D76AC();
+    if (res < 0)
+    {
+        FUN_02032F20((u32)res);
+        FUN_02031D20(FUN_02032E8C, 0);
+        return;
+    }
+    else if (res > 0)
+    {
+        FUN_02031D20(FUN_02033060, 0);
+    }
+}
+
+THUMB_FUNC void FUN_0203341C()
+{
+    if (FUN_0202E4C8() != 0)
+    {
+        FUN_02016828(3, 0x31, 0x31000);
+        if (FUN_0202F918(1, 1, 0x200, 1) != 0)
+        {
+            MOD05_021D74E0(UNK_021C5A00->unk28, 0x31);
+            MOD04_021D83A8(FUN_020335D0);
+            FUN_0202FA10();
+            FUN_02031D20(FUN_020333F0, 0);
+        }
+    }
+}
+
+THUMB_FUNC void *FUN_0203346C(struct SaveBlock2 *sav2, u32 param1)
+{
+    Sav2_PlayerData_GetProfileAddr(sav2);
```

Comment:
this seems a bit strange being left here on it's own, almost like it should feed into something as a parameter

Reply:
It's strange so I double checked.

This function starts as
```asm
	thumb_func_start FUN_0203346C
FUN_0203346C: ; 0x0203346C
	push {r3-r5, lr}
	add r5, r0, #0x0
	add r4, r1, #0x0
	bl Sav2_PlayerData_GetProfileAddr
	bl FUN_02030F40
	cmp r0, #0x0
```
So the only possible use of the result of `Sav2_PlayerData_GetProfileAddr` is as paramter to `FUN_02030F40` which was already decomiled and takes no argument.

But let's check its asm to be sure:
```asm
	thumb_func_start FUN_02030F40
FUN_02030F40: ; 0x02030F40
	push {r3, lr}
	ldr r0, _02030F5C ; =UNK_021C59F4
	ldr r0, [r0, #0x4]
	cmp r0, #0x0
	beq _02030F56
	bl FUN_02033534
	cmp r0, #0x13
	blt _02030F56
	mov r0, #0x1
	pop {r3, pc}
_02030F56:
	bl FUN_0202EDF8
	pop {r3, pc}
	.balign 4
_02030F5C: .word UNK_021C59F4
```
It starts by overriding the value of r0 so this function really takes no argument for sure.

I guess `Sav2_PlayerData_GetProfileAddr(sav2);` is just here by mistake...